src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/AncientAltarListener.java

```diff
@@ -156,8 +156,18 @@ private void usePedestal(@Nonnull Block pedestal, @Nonnull Player p) {
             Slimefun.runSync(() -> removedItems.remove(uuid), 30L);
 
             entity.remove();
-            p.getInventory().addItem(pedestalItem.getOriginalItemStack(entity));
             p.playSound(pedestal.getLocation(), Sound.ENTITY_ITEM_PICKUP, 1F, 1F);
+
+            /*
+             * Fixes #3476
+             * Drop the item instead if the player's inventory is full and
+             * no stack space left else add the item in the inventory
+             */
+            for (Map.Entry<Integer, ItemStack> stackEntry : p.getInventory().addItem(pedestalItem.getOriginalItemStack(entity)).entrySet()) {
+                if (stackEntry != null) {
```

Comment:
I’m fairly sure when iterating the entry set you can’t get a nullable entry? Could be wrong as I’m not at my desk but I feel like this isn’t needed?

Reply:
Also, may just be best to iterate the values directly, don’t think the int it required in this context at all.

Reply:
> I’m fairly sure when iterating the entry set you can’t get a nullable entry? Could be wrong as I’m not at my desk but I feel like this isn’t needed?
Yes thats correct, I added it for debugging and safety ig when I was testing it ingame, I can fully remove it in the next commit